common/queue.h

```diff
@@ -86,6 +87,72 @@ class Queue {
 	List<T>	_impl;
 };
 
+/**
+ * Queue ordered by a provided priority function
+ * NOTE: Unlike in the C std library, we have to provide a comparator that sorts
+ * the array, so that the smallest priority comes last
+ */
```

Comment:
Thinking about this a bit more, this PR should probably be put on hold until this comment can be addressed.